### PR TITLE
DOC: Add example to lstm doc

### PR DESCRIPTION
**Pull request by sinhrks**
Added some explanations to ``functions.lstm`` which I felt little difficult to use.

Rendered output:

lstm

### CODE REVIEW

**Comment by beam2d** [comment]
Thank you for improving the doc! I will check it.

File: chainer/functions/lstm.py [hunk 1]
```diff
@@ -152,9 +152,20 @@ def lstm(c_prev, x):
...
-    Then, first the incoming signal :math:`x` is split along the second
-    dimension into four arrays :math:`a, i, f, o` of the same shapes. Second,
-    it computes outputs as:
+
+    First, the incoming signal :math:`x` is split into four arrays
+    :math:`a, i, f, o` of the same shapes along the second axis.
+    It means that :math:`x` 's second axis must have 4 times the length of
+    :math:`c_{\\text{prev}}`.
+
+    Splitted input signals are corresponding to:
```
Comment (beam2d) [inline chainer/functions/lstm.py]: The splitted input signals are corresponding to

File: chainer/functions/lstm.py [hunk 2]
```diff
@@ -179,5 +190,21 @@ def lstm(c_prev, x):
     `Long Short-Term Memory in Recurrent Neural Networks \
     <http://www.felixgers.de/papers/phd.pdf>`_.
 
+    .. admonition:: Example
+
+        Assuming ``y`` is current input signal, ``c`` is previous cell state,
+        and ``h`` is previous output signal from ``lstm`` layer with all the
+        length are ``n_units``. Most typical prepareation of ``x`` is:
```
Comment (beam2d) [inline chainer/functions/lstm.py]: -> Assuming ``y`` is **the** current input signal, ``c`` is **the** previous cell state, and ``h`` is **the** previous output signal from **an** ``lstm`` **function**. **Each of ``y``, ``c``, and ``h`` has ``n_units`` channels**. Most typical **preparation** of ``x`` is:

File: chainer/functions/lstm.py [hunk 3]
```diff
@@ -179,5 +190,21 @@ def lstm(c_prev, x):
...
+
+        >>> model = FunctionSet(w=F.Linear(n_units, 4 * n_units),
+        ...                     v=F.Linear(n_units, 4 * n_units),
+        ...                     ...)
+        >>> x = model.w(y) + model.v(h)
+        >>> c, h = F.lstm(c, x)
+
+        It corresponts to calcurate each input sources :math:`a, i, f, o` using
+        different weight matrices and biases for current input ``y``
+        and previous output ``h``.
```
Comment (beam2d) [inline chainer/functions/lstm.py]: -> It **corresponds** to **calculate** **the** input sources :math:`a, i, f, o` **from** **the** current input ``y`` and **the** previous output ``h``. **Different parameters are used for different kinds of input sources.**

**Comment by beam2d** [comment]
I added some suggestions for the expressions. It includes typos which must be fixed, while other suggestions may be ignored if you do not think a fix is needed. (I am not a native, so my suggestion may contain errors :))

**Comment by sinhrks** [comment]
Thanks for review. Fixed based on your points, as I'm not also:)

**Comment by beam2d** [comment]
[LGTM](http://lgtm.in/i/0IWb1Ndtn)

**Comment by beam2d** [comment]
Thank you!

**Pull request merged**